src/system/lt.h

```diff
@@ -46,19 +46,21 @@ static inline void *lt_push(Lt *lt, void *res, Dtor dtor)
 {
     trace_assert(lt);
 
-    if (lt->size >= lt->capacity) {
+    if ((size_t)(lt->slots_end - lt->slots) >= lt->capacity) {
         if (lt->capacity == 0) {
             lt->capacity = LT_INITIAL_CAPACITY;
             lt->slots = calloc(LT_INITIAL_CAPACITY, sizeof(Slot));
+            lt->slots_end = lt->slots;
         } else {
             lt->capacity *= 2;
             lt->slots = realloc(lt->slots, lt->capacity * sizeof(Slot));
+            lt->slots_end = lt->slots;
```

Comment:
You probably don't want to reset `slots_end` back to the beginning after the reallocation.